src/android/BrotherPrinter.java

```diff
@@ -290,6 +297,20 @@ private void sendDiscoveredPrinters(final CallbackContext callbackctx, List<Disc
         callbackctx.sendPluginResult(result);
     }
 
+    private static final int OPEN_BLUETOOTH_SETTINGS_REQUEST = 0x2345;
+    private void pairBluetoothPrinters(final CallbackContext callbackctx) {
+        cordova.getThreadPool().execute(new Runnable() {
+            @Override
+            public void run() {
+                final Intent openPairSection = new Intent();
+                openPairSection.setAction(Settings.ACTION_BLUETOOTH_SETTINGS);
+                cordova.startActivityForResult(BrotherPrinter.this, openPairSection, OPEN_BLUETOOTH_SETTINGS_REQUEST);
+
+                callbackctx.success();
+            }
+        });
+    }
+
```

Comment:
Can't we just use [BluetoothManager.getAdapter()](https://developer.android.com/reference/android/bluetooth/BluetoothManager.html#getAdapter()) to get [BluetoothAdapter](https://developer.android.com/reference/android/bluetooth/BluetoothAdapter.html) along with [BluetoothDevice](). Listen for: [ACTION_DISCOVERY_STARTED](https://developer.android.com/reference/android/bluetooth/BluetoothAdapter.html#ACTION_DISCOVERY_STARTED), [ACTION_DISCOVERY_FINISHED](https://developer.android.com/reference/android/bluetooth/BluetoothAdapter.html#ACTION_DISCOVERY_FINISHED), and [ACTION_FOUND](https://developer.android.com/reference/android/bluetooth/BluetoothDevice.html#ACTION_FOUND). Then call [BluetoothAdapter.startDiscovery()](https://developer.android.com/reference/android/bluetooth/BluetoothAdapter.html#startDiscovery()).

Then when you find a brother device via ACTION_FOUND, simply call: [BluetoothAdapter.getRemoteDevice()](https://developer.android.com/reference/android/bluetooth/BluetoothAdapter.html#getRemoteDevice(java.lang.String)) to get a BluetoothDevice. Then on the device returned from getRemoteDevice(), issue a createBond() call: [BluetoothDevice.createBond()](https://developer.android.com/reference/android/bluetooth/BluetoothDevice.html#createBond()).

Reply:
alternatively... this may provide some insight too, though probably overkill for what we need here https://github.com/tanelih/phonegap-bluetooth-plugin this is an android-only plugin for old phonegap, but the android bits should be usable.

Reply:
we can make this better on android with a new ticket though... if you want to stick to IOS here. lemme know...

Reply:
We could probably use `BluetoothManager` instead of `BluetoothAdapter` without much issue.  It seems like the code would operate roughly the same without it.

As far as the `.startDiscovery()` call, that would seem to imply you'd like to do the pairing within the plugin itself instead of externally.  Is this what you're intending? The only thing that really stopped me from doing that before is that I don't know of any UI we can display to the user that's similar to `iOS`'s.  As such, we'd either have to create that UI, or forward the options through the plugin. I didn't like either option really, because the first forces a UI we design on the developer, and the second means that there's different behavior between the plugins.

Unless you're saying you just want to find the first printer you can find and try to bond with it immediately.  The only problem with that is when there are multiple printers and you only want to pair with a specific one.

Reply:
> you'd like to do the pairing within the plugin itself

ideally, yes... the plugin should encapsulate all of the steps to provision a printer. that's why it's possible to send USB commands. for example, to set the printer's network settings. i'd like to support the same sort of functionality for bluetooth printing too, if possible.

however, that would mean that it should be done similarly on the IOS side too...  and that may be problematic.

i think CoreBluetooth should do what we need, on the IOS side, to get a list of printers and enumerate the services, pair, etc? here's a nice wrapper for it: https://github.com/LGBluetooth/LGBluetooth but i'm not 100% sure if it's possible to pair with "legacy" bluetooth devices programmatically on IOS. are these printers using BLE? if so, wonderful, if not... damnit!

> The only thing that really stopped me from doing that before is that I don't know of any UI we can display to the user

well, we don't need a UI, necessarily. this is just a cordova plugin. it's exposing an api to app developers to do with as they please. we can provide an example implementation with a UI for test or example purposes, but that's ultimately a burden that falls on the end user of the plugin, to implement as needed, for their specific purposes.

ideally we just spit back a list of compatible printers, and then the implementor can show that list to a user and let them pick the one they want, or their app code can pick one automatically based on whatever criteria they want. that's outside the scope of this plugin, i think. there should be no UI here, just API.

this sort of workflow is already there for network printing. you ask for a list of printers, then pick one. ideally we can do this on both android and ios for bluetooth printing as well. but, i do understand if it's not possible on IOS. i'm sure there are "ways" such as using private frameworks, and for enterprise usage, that's fine, but you can't use private frameworks if you're pushing to the app store. for my particular use cases, app store compatibility is irrelevant, but i don't know about everyone else.

that being said... 

[PrivateFrameworks.BluetoothManager](https://github.com/nst/iOS-Runtime-Headers/blob/master/PrivateFrameworks/BluetoothManager.framework/BluetoothManager.h) has the following methods: `scanForConnectableDevices()` and `connectDevice()`.

maybe we can add a config.xml flag to include those methods into the ios side of the plugin, optionally, depending whether they want to fully automate, with the caveat that it's only for enterprise or private use, not for deploying to the app store.

thoughts?